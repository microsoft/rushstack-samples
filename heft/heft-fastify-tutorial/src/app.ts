import * as path from 'path';
import * as ejs from 'ejs';
import { FastifyPluginAsync, FastifyInstance } from 'fastify';
import PointOfView from 'point-of-view';
import FastifyStatic from 'fastify-static';

import { HomeController } from './web/HomeController';
import { BooksController } from './web/books/BooksController';
import { ApiBooksController } from './api/books/ApiBooksController';

const app: FastifyPluginAsync = async (fastify: FastifyInstance) => {
  // Register plugins
  await fastify.register(PointOfView, {
    engine: {
      ejs: ejs
    },
    layout: 'layouts/global.ejs',
    root: path.join(__dirname, './web')
  });
  await fastify.register(FastifyStatic, {
    root: path.join(__dirname, './assets'),
    prefix: '/assets'
  });

  // Application routes, organized into small "controllers"
  await fastify.register(HomeController.register, { prefix: '/' });
  await fastify.register(BooksController.register, { prefix: '/books' });
  await fastify.register(ApiBooksController.register, { prefix: '/api/books' });
};

// Our `app.ts` exports a Fastify Plugin, which means you can run it directly
// with the fastify-cli after compiling:
//
//   fastify lib/app.js
//
// Doing this also keeps unit tests clean, as they can import the app definition
// without getting in the mud with the specifics of starting the server.
//
// In production, you'll want to be able to configure Fastify (for example,
// to specify where output logs should go), so we would run `server.ts` instead:
//
//   node lib/server.js
//
export default app;
