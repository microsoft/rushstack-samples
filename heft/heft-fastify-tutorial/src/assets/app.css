body,
html {
  background-color: #be5f3e;
  font-family: Helvetica, Verdana, Arial, sans-serif;
  margin: 0px;
  padding: 0px;
}

.site {
  background-color: #fffff0;
  border-radius: 8px;
  color: #333;
  padding: 16px;
  margin: 16px;
}

h2 {
  margin-top: 0px;
  border-top: 0px;
}

table.books {
  border-collapse: collapse;
}

table.books td,
table.books th {
  border: solid 1px #666;
  padding: 0.25em 0.5em;
}
