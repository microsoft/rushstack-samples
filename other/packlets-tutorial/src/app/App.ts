// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { DataModel, ExampleModel } from '../packlets/data-model';

import { Logger, MessageType } from '../packlets/logging';
import { MainReport } from '../packlets/reports';

export class App {
  public run(): void {
    const logger: Logger = new Logger();
    logger.log(MessageType.Info, 'Starting app...');

    const dataModel: DataModel = new ExampleModel(logger);
    const report: MainReport = new MainReport(logger);
    report.showReport(dataModel);

    logger.log(MessageType.Info, 'Operation completed successfully');
  }
}
