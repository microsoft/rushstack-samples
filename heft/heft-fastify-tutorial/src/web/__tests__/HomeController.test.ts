import fastify from 'fastify';
import app from '../../app';

describe('HomeController', () => {
  describe('index', () => {
    it('renders an HTML index page', async () => {
      const subject = fastify().register(app);
      const response = await subject.inject({
        method: 'GET',
        url: '/'
      });

      expect(response.statusCode).toEqual(200);
      expect(response.body).toMatchSnapshot();
    });
  });
});
