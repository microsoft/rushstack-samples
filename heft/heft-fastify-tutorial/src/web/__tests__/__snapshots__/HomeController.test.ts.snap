// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeController index renders an HTML index page 1`] = `
"<html>
<head>
<link rel=\\"stylesheet\\" href=\\"/assets/app.css\\">
</head>
<body>
  <div class=\\"site\\">
    <h2>Book World</h2>

Welcome to Book World!

See our <a href=\\"/books\\">list of books</a>, or try our <a href=\\"/api/books\\">REST API</a>.

  </div>
</body>
</html>
"
`;
